Services/UI/OJS.Services.Ui.Business/Implementations/ContestResultsAggregatorService.cs

```diff
@@ -19,13 +26,18 @@ public ContestResultsAggregatorService(IParticipantsCommonDataService participan
             bool isFullResults,
             bool isExportResults = false)
         {
+            var contestActivityEntity = this.activityService
+                .GetContestActivity(contest.Map<ContestForActivityServiceModel>())
+                .GetAwaiter()
```

Comment:
Is there any way to make this method async to bypass .GetAwaiter().GetResult() or this line is necessary here?

Reply:
I initially tried to make it async but it was throwing errors in some cases when multiple calls were being made. After discussion I kept this implementation